packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -0,0 +1,241 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualNode } from './virtual-node';
+
+/**
+ * Properties for the Weighted Targets in the route
+ */
+export interface WeightedTarget {
+  /**
+   * The VirtualNode the route points to
+   */
+  readonly virtualNode: IVirtualNode;
+
+  /**
+   * The weight for the target
+   *
+   * @default 1
+   */
+  readonly weight?: number;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface HttpRouteMatch {
```

Comment:
This is exactly the same as the existing `HttpRouteMatch` interface. Perhaps we should have only one interface for this?

Reply:
This is actually only the case because we have not implemented the full `RouteMatch` for routes yet:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-route-httproutematch.html

They do differ on the underlying CFN resource between gateway routes and routes. This could change in the future, but right now it is the case. I don't see a way this could be handled with a single `HttpRouteMatch` interface.

Reply:
OK. I guess we changed something in the other PR to not use the name `HttpRouteMatch` anymore...?

We can't have 2 interfaces exported with the same name from one module.

Reply:
The other one is called `HttpGatewayRouteMatch` :)